File: python/p041.py
import utils

primes = utils.primes(7654321)

def p(n):
    sn = str(n)
    lsn = len(sn)

    if lsn > 10:
        return False
    
    return set([ int(d) for d in sn ]) == set(range(1, len(sn) + 1))


primes.reverse()
for prime in primes:
    if p(prime):
        print prime
        break

File: erlang/src/p041.erl
-module(p041).

-export([start/0]).

f() ->
  Primes = utils:primes(7654321),
  hd(lists:dropwhile(fun(N) -> not utils:pandigital(N) end, lists:reverse(Primes))).

start() ->
  io:format("~p ~n", [f()]).
